packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -30,7 +16,7 @@ export interface PanelChromeProps {
   children: (innerWidth: number, innerHeight: number) => ReactNode;
   padding?: PanelPadding;
   title?: string;
-  titleItems?: PanelChromeInfoState[];
+  titleItems?: (innerWidth: number, innerHeight: number) => ReactNode;
```

Comment:
why does this need innerWidth and innerHieght? (and innerWidth and innerHeight of what, header, panel content?,)

Reply:
but I think it could be a good idea to make this a more extensible hook, I think PanelChrome might need to be very generic and support a lot of different flavors / features (as we explore more use cases from Scenes framework)

Reply:
>why does this need innerWidth and innerHieght?

The idea is that the space _each_ "item" is allowed is determined by the `PanelChrome` and passed down to the component responsible for setting up the items themselves (in our case the `PanelHeaderTitleItems`). The innerWidth and innerHeight here are probably better named infoNodeHeight and infoNodeWidth or something of the kind. I can see they are not yet used in PanelHeaderTitleItems but will set the 32x32 px square we have around each icon:
![image](https://user-images.githubusercontent.com/13227501/209099949-b49fafeb-0a68-4f40-8b71-ce47497ecb47.png)

Reply:
Not sure I understand still , seems unnecessary?

Reply:
I can't see it being used in https://github.com/grafana/grafana/blob/cf99a137932cd1126b166a3bb5e449c92dfece6f/public/app/features/dashboard/dashgrid/PanelHeader/PanelHeaderTitleItems.tsx

Reply:
Indeed is not being used in the current implementation and is not needed.

When we did the analysis, we thought we could bring consistency to the rendering of the `titleItems` by offering the width and height, like saying "32px of height is what your items should have". However, when I was implementing this, I noticed how complicated it is to do that. For example, I had to pass the `innerWidth` and `innerHeight` down to 3 different components in order to render the Panel notices with those dimensions (PanelHeaderTitleItems --> PanelHeaderNotices --> PanelHeaderNotice).

I would say that this was a failed attempt to bring control and consistency to the approach of having `title items` as a render call (the most significant disadvantage of this approach is that we don't have control over what the developer will place in it, but it is also a big advantage considering we are creating a component that can be used outside of the Grafana core domain).

I will remove the exposure of these dimensions and add the task of adding to the PanelChrome MDX documentation that `title items at least should have 32px as their height`.

Reply:
> I will remove the exposure of these dimensions and add the task of adding to the PanelChrome MDX documentation that title items at least should have 32px as their height.

IMO the sizing of the title items should be set via wrapper html element that renders it - why would someone using the API have to set the dimensions of an action item? It's an easy path for creating inconsistencies in the panel chrome.

Reply:
Thanks a lot for the feedback 🙌🏾 , after our discussion I have made changes that tackle this point:

- `PanelChrome` now supports standard props such as `description, links, and panelNotices` 
- `titleItems` is now an array of React nodes that will render edge cases like `alerts and timeshift`